api/src/main/java/org/apache/gravitino/stats/Statistic.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.stats;
+
+import java.util.Optional;
+import org.apache.gravitino.annotation.Evolving;
+
+/**
+ * Statistic interface represents a statistic that can be associated with a metadata object. It can
+ * be used to store various types of statistics, for example, table statistics, partition
+ * statistics, fileset statistics, etc.
+ *
+ * <p>For different statistics, Gravitino will use different storage solution for them. Table
+ * statistics will be stored in Gravitino metadata store, usually the Gravitino's metadata is
+ * relational database, it's difficult to store millions rows of records.
+ *
+ * <p>But for partition statistics, there may be millions of partitions in a large table. So
+ * Gravitino will use a external storage solution. Gravitino will use a file to store all
+ * partitions. The partition statistics may be several megabytes. If a partition is stored as a
+ * file, it will cause millions of small files in the storage system. It will cause high pressure
+ * for HDFS if you choose HDFS as external storage. If the partition statistic is requested for
+ * multiple times, Gravitino will use cache to improve performance.
```

Comment:
You don't have to explain the implementation here. The javadoc here is mainly for the users about the functionality and how to use about the `statistic`.

Reply:
OK, I can remove them later. Just to let reviewers to realize of my solution.